### DISCUSSION TITLE
Property based testing

### DISCUSSION THREAD

**Discussion by Jaxing** [dbt-labs/dbt-core#5745] [Ideas]
Hey,
using the tests in dbt is super helpful to figure out if your query works with your data but it can become very heavy to run and it still might not test every possible case for your code.

[Property based testing](https://fsharpforfunandprofit.com/posts/property-based-testing/) is a method for writing unit tests that doesn't test specific inputs but instead generate random input (could be based on the column types of a table) and makeing sure that a function (or query) fullfills a property (e.g. idempotents or commutativity (incase of incremental queries) etc).

Would love to discuss with people if we could create a package for dbt that could do this!

**Comment by JamesG419** [comment]
@Jaxing this sounds like something really useful. Interested in helping with its on your mind
